File: CPP/119_Pascal's_Triangle_II.cc
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> ans;
        ans.push_back(1);
        if(rowIndex==0)return ans;
        if(rowIndex==1){
            ans.push_back(1);
            return ans;
        }
        vector<int> cur=getRow(rowIndex-1);
        for(int j = 1;j < rowIndex;j++)
        {
              ans.push_back(cur[j] + cur[j - 1]);
        }
        ans.push_back(1);
        
        return ans;
    }
};

File: Java/119_Pascal's_Triangle_II.java
public class Solution {
    // 2 ms
    public List<Integer> getRow(int rowIndex) {
        List<Integer> res = new ArrayList<>();
        res.add(1);
        if(rowIndex<=0)return res;
        List<Integer> pre = getRow(rowIndex-1);
        int p = pre.get(0), q;
        for(int i=1;i<pre.size();i++){
            q=pre.get(i);
            res.add(p+q);
            p=q;
        }
        res.add(1);
        return res;
    }
}